### ISSUE TITLE
Saving throws an error when atom-beautify is enabled

### ISSUE THREAD

**Issue by ejpcmac** [dschwen/switch-header-source#29]
### Steps to reproduce

1. Enable formatting on save in `atom-beautify`, with `switch-header-source` installed
2. Edit and save a C file

### Unexpected behaviour

An error is thrown. It was not the case with previous versions.

### Expected behaviour

No error should be thrown.

### Environment

**Atom**: 1.25.0 x64
**Electron**: 1.7.11
**OS**: "Fedora release
**Thrown From**: [switch-header-source](https://github.com/dschwen/switch-header-source) package 0.28.0

### Stack Trace

Uncaught Error: ENOENT: no such file or directory, lstat '/***/.atom-beautify.***.c'

```
At fs.js:961

Error: ENOENT: no such file or directory, lstat '/***/.atom-beautify.***.c'
    at fs.lstatSync (fs.js:961:11)
    at Object.fs.lstatSync (ELECTRON_ASAR.js:244:16)
    at Proxy.realpathSync (fs.js:1641:21)
    at Proxy.fs.realpathSync (ELECTRON_ASAR.js:321:29)
    at Object.switchMapDelete (/packages/switch-header-source/lib/switch-header-source.coffee:80:34)
    at /packages/switch-header-source/lib/switch-header-source.coffee:119:12
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom/resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at didChangeCallback (/usr/share/atom/resources/app/src/project.js:357:28)
    at PathWatcher.onNativeEvents (/usr/share/atom/resources/app/src/path-watcher.js:539:13)
    at sub.native.onDidChange.events (/usr/share/atom/resources/app/src/path-watcher.js:464:64)
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom/resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at NativeWatcher.onEvents (/usr/share/atom/resources/app/src/path-watcher.js:325:24)
    at handler (/usr/share/atom/resources/app/src/path-watcher.js:133:13)
```

### Commands

```
     -0:06.3.0 vim-mode-plus:insert-below-with-newline (input.hidden-input)
     -0:05.6.0 intentions:highlight (input.hidden-input)
     -0:05.5.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.32.2 
atom-clock 0.1.16 
atom-ide-ui 0.9.4 
atom-updater-linux 1.1.2 
auto-detect-indentation 1.3.0 
autoclose-html 0.23.0 
autocomplete-clang 0.11.4 
autocomplete-cmake 0.3.1 
autocomplete-racer 0.1.3 
busy 0.7.0 
busy-signal 1.4.3 
dbg 1.6.3 
dbg-gdb 1.7.8 
editorconfig 2.2.2 
file-icons 2.1.18 
git-diff-details 1.4.0 
git-time-machine 1.5.9 
gitignore-snippets 0.2.5 
highlight-selected 0.13.1 
ide-crystal 0.3.3 
ide-elixir 0.2.19 
ide-flowtype 0.22.1 
ide-rust 0.15.1 
intentions 1.1.5 
language-cmake 1.1.0 
language-crystal-actual 0.37.2 
language-diff 0.7.0 
language-doxygen 0.7.0 
language-elixir 0.22.0 
language-erlang 3.2.0 
language-generic-config 1.4.0 
language-gettext 0.6.1 
language-gitignore 0.3.0 
language-ini 1.19.0 
language-latex 1.1.1 
language-llvm-ir 1.1.1 
language-rust 0.4.12 
language-textile 0.2.0 
latex 0.49.0 
latexer 0.3.0 
linter-chktex 1.3.1 
linter-clang 4.1.2 
linter-coffeelint 1.3.1 
linter-mixed-indent 1.1.1 
linter-pylint 2.1.1 
linter-ruby 1.2.6 
local-history 4.3.1 
minimap 4.29.8 
minimap-find-and-replace 4.5.2 
minimap-git-diff 4.3.1 
next-line-comment 0.2.1 
output-panel 0.3.4 
pdf-view 0.67.0 
phoenix-elixir-snippets 0.4.0 
platformio-ide-terminal 2.8.1 
process-palette 0.16.1 
show-invisibles-plus 0.3.1 
Sublime-Style-Column-Selection 1.7.4 
switch-header-source 0.28.0 
sync-settings 0.8.6 
textile-preview 0.1.2 
theme-switch 0.6.0 
todo-show 2.3.0 
tool-bar 1.1.7 
trailing-spaces 0.4.0 
vim-mode-plus 1.32.0 
```

**Comment by dschwen** [comment]
Looks like a race condition where beautify weird a temporary file that the switch header source plugin notices. When the shs plugin then calls realPathSync the for is gone already. I'll catch that error.

**Comment by dschwen** [comment]
I just published `v0.28.1`. Could you guys check if that fixes the issue?

**Comment by ejpcmac** [comment]
@dschwen Yes it seems. Thank you very much!
